tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -778,6 +780,65 @@ public function testConnectionParamsArePassedToTheQueryCacheProfileInExecuteCach
     }
 
     /**
+     * @group DBAL-2821
+     */
+    public function testShouldNotPassPlatformInParamsToTheQueryCacheProfileInExecuteCacheQuery(): void
+    {
+        $resultCacheDriverMock = $this->createMock(Cache::class);
+
+        $resultCacheDriverMock
+            ->expects($this->atLeastOnce())
+            ->method('fetch')
+            ->with('cacheKey')
+            ->will($this->returnValue(['realKey' => []]));
+
+        /* @var $queryCacheProfileMock QueryCacheProfile|\PHPUnit_Framework_MockObject_MockObject */
+        $queryCacheProfileMock = $this->createMock(QueryCacheProfile::class);
+
+        $queryCacheProfileMock
+            ->expects($this->any())
+            ->method('getResultCacheDriver')
+            ->will($this->returnValue($resultCacheDriverMock));
+
+        $query  = 'SELECT 1';
+
+        $connectionParams = $this->params;
+
+        // This is our main expectation
```

Comment:
Comment can be dropped: if there were no expectations, the test would be marked as risky